src/transmission.rs

```diff
@@ -21,6 +21,9 @@ use crate::events::{Events, EventsResponse};
 use crate::response::{HoneyResponse, Response};
 use crate::sender::Sender;
 
+// Re-export reqwest client to help users avoid versioning issues.
+pub use reqwest::{Client as HttpClient, ClientBuilder as HttpClientBuilder};
```

Comment:
I'd prefer to avoid this in case we want to move away from reqwest. (There's already a PR with the intent of using surf as well, so I'd rather not expose this at all if possible)

Reply:
sounds good! just reverted the public interface changes.

in case this is helpful: this issue has come up a few times for my [oauth2](https://crates.io/crates/oauth2) crate as well. folks kept filing issues wanting to use different HTTP clients/versions. I ended up making the various request interfaces generic to satisfy everyone:
```
pub fn request<F, RE>(
    self,
    http_client: F
) -> Result<TR, RequestTokenError<RE, TE>>
where
    F: FnOnce(HttpRequest) -> Result<HttpResponse, RE>,
    RE: Error + 'static, 
```
and the async version:
```
pub async fn request_async<C, F, RE>(
    self,
    http_client: C
) -> Result<TR, RequestTokenError<RE, TE>>
where
    C: FnOnce(HttpRequest) -> F,
    F: Future<Output = Result<HttpResponse, RE>>,
    RE: Error + 'static, 
```

looks like there's this [http-client](https://crates.io/crates/http-client) crate that Surf uses as an abstraction layer. I might migrate my crates over to that now that it exists, and just provide a reqwest adapter 🤔